bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.crypt.capi;
+
+import com.mongodb.crypt.capi.CAPI.mongocrypt_binary_t;
+import com.sun.jna.Memory;
+import com.sun.jna.Pointer;
+import org.bson.BsonBinaryWriter;
+import org.bson.BsonDocument;
+import org.bson.RawBsonDocument;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistries;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.io.BasicOutputBuffer;
+
+import java.nio.ByteBuffer;
+
+import static com.mongodb.crypt.capi.CAPI.mongocrypt_binary_data;
+import static com.mongodb.crypt.capi.CAPI.mongocrypt_binary_len;
+import static com.mongodb.crypt.capi.CAPI.mongocrypt_binary_new_from_data;
+
+final class CAPIHelper {
+
+    private static final CodecRegistry CODEC_REGISTRY = CodecRegistries.fromProviders(new BsonValueCodecProvider());
+
+    @SuppressWarnings("unchecked")
+    static mongocrypt_binary_t toBinary(final BsonDocument document) {
+        BasicOutputBuffer buffer = new BasicOutputBuffer();
+        BsonBinaryWriter writer = new BsonBinaryWriter(buffer);
+        ((Codec<BsonDocument>) CODEC_REGISTRY.get(document.getClass())).encode(writer, document, EncoderContext.builder().build());
+
+        Pointer pointer = new Memory(buffer.size());
```

Comment:
`finalize` is protected, so it can't be called directly by the library.  It's unfortunate that JNA relies on finalize rather than AutoCloseable, but c'est la vie.